content/blog/2019-09-23-grpc-on-dotnetcore.md

```diff
@@ -0,0 +1,61 @@
+---
+author: Sourabh Shirhatti
+author-link: https://twitter.com/sshirhatti
+date: "2019-09-23T00:00:00Z"
+published: true
+title: .NET Core ❤ gRPC
+url: blog/grpc-on-dotnetcore
+---
+
+_This is a guest post by [Sourabh Shirhatti](https://twitter.com/sshirhatti), a Program Manger on the .NET team at Microsoft._
+
+The .NET team at Microsoft has been working in close collaboration with the gRPC team since November 2018 on a new fully managed implementation of gRPC for .NET Core.
+
+We're pleased to announce that **grpc-dotnet** is now available with .NET Core 3.0 today!
+
+## How to get it?
+
+**grpc-dotnet** packages have just been released to [NuGet.org](https://www.nuget.org/profiles/grpc-packages) and are already available for use in your projects. These packages also require the latest .NET Core 3.0 shared framework. You can download the .NET Core 3.0 SDK for your dev machine and build servers from the [.NET Core 3.0 download page](https://aka.ms/netcore3download) to acquire the shared framework. 
+
+## Getting Started
+
+gRPC templates are included as part of the .NET SDK. To get started, navigate to a console window after installing the SDK and run the following commands.
+
+```sh
+dotnet new grpc -o GrpcGreeter
+cd GrpcGreeter
+dotnet run
+```
+
+To create a gRPC client and test with the newly created gRPC Greeter service, you can [follow the rest of this tutorial here](https://docs.microsoft.com/aspnet/core/tutorials/grpc/grpc-start).
+
+## Doesn't gRPC already work with .NET Core?
+
+There are currently two official implementations of gRPC for .NET:
+
+- [**Grpc.Core**](https://github.com/grpc/grpc/tree/master/src/csharp): The original gRPC C# implementation based on the native gRPC Core library.
+- [**grpc-dotnet**](https://github.com/grpc/grpc-dotnet): The new implementation written entirely in C# with no native dependencies and based on the newly released .NET Core 3.0.
+
+The implementations coexist side-by-side and each has its own advantages in terms of available features, integrations, supported platforms, maturity level and performance. Both implementations share the same API for invoking and handling RPCs, thus limiting the lock-in and enabling users to choose the implementation that satisfies their needs the best.
+
+##  What's new?
+
+Unlike the existing C-Core based implementation ([Grpc.Core](https://github.com/grpc/grpc/tree/master/src/csharp)), the new libraries ([grpc-dotnet](https://github.com/grpc/grpc-dotnet)) make use of the existing networking primitives in the .NET Core Base Class Libraries (BCL). The diagram below highlights the difference between the existing **Grpc.Core** library and the new **grpc-dotnet** libraries.
```

Comment:
optional nit: "...make use of the existing networking primitives in the .NET Core Base Class Libraries (BCL)"  for more context, also mention that grpc-dotnet relies on the newly added HTTP/2 support in .NET Core 3 which is also the reason why something like this wasn't possible before?

Reply:
It felt unnatural adding it based on how the blog is currently structured, so I've omitted this.